enclave-tls/src/enclave_quotes/internal/etls_enclave_quote_select.c

```diff
@@ -34,6 +35,23 @@ enclave_tls_err_t etls_enclave_quote_select(etls_core_context_t *ctx,
 	for (unsigned int i = 0; i < registerd_enclave_quote_nums; ++i) {
 		if (type && strcmp(type, enclave_quotes_ctx[i]->opts->type))
 			continue;
+
+		if (enclave_quotes_ctx[i]->opts->flags) {
+                        bool sgx_supported = (is_sgx2_supported() || is_sgx1_supported());
+                        if (!sgx_supported) {
+                                ETLS_ERR("Selecting enclave quotes while lacking sgx \
+                                 capability\n");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                } else {
+                        bool sgx_device_exists = (is_dcap_oot_kernel_driver() ||
+                                        is_legacy_oot_kernel_driver() || is_in_tree_kernel_driver());
+                        if (!sgx_device_exists) {
+                                ETLS_ERR("Selecting enclave quotes while \
+                                                sgx device file doesn't exist\n");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                }
```

Comment:
I initially reviewed the tls_tls_wrapper_select() so I see here is another duplicated logic. Please move this chunk to a common function.

Reply:
ditto.
And abstract a common function